src/repository/UserDatabaseRepository.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+import type { Database, TokenObject, User, UserCredentials } from '../types';
+
+import PasswordHasher from '../lib/PasswordHasher';
+import HttpError from '../lib/HttpError';
+
+class UserDatabaseRepository {
+  _collection: Object;
+
+  constructor(database: Database) {
+    this._collection = database.getCollection('users');
+  }
+
+  createWithCredentials = async (userCredentials: UserCredentials): Promise<User> => {
+    const { username, password } = userCredentials;
+
+    const salt = await PasswordHasher.generateSalt();
+    const passwordHash = await PasswordHasher.hash(password, salt);
+    const modelToSave = {
+      accessTokens: [],
+      created_at: new Date(),
+      created_by: null,
+      passwordHash,
+      salt,
+      username,
+    };
+
+    const user = (await this._collection.insert(
+      modelToSave,
+      { fullResult: true },
+    ))[0];
+    return { ...user, id: user._id.toString() };
```

Comment:
this should be in some kind of Translator, but since we need it only in a few places inside UserRepo and only for id, i think its okey then.